### ISSUE TITLE
`Next` keyword causes parse failure

### ISSUE THREAD

**Issue by sjbarag** [sjbarag/brs#145]
The Reference BrightScript Implementation (RBI) allows the `next` keyword to be used as a synonym for `end for` as best I can tell.  `brs` doesn't support that right now, but it definitely should

## Broken snippet:

```brightscript
for each num in [1,2,3]
    print num
next
```

Originally reported by @twitchbronbron.  Thanks friend!

**Comment by sjbarag** [comment]
@sjbarag be sure to figure out if `next` can be used like `continue` in other langues!  e.g.

```brightscript
for each foo in [1, 2, 3, 4, 5]
    if foo < 3 then next
    print foo
next
```


Is this valid?  If so, does it only print `3`, `4`, and `5`?

**Comment by sjbarag** [comment]
> @sjbarag be sure to figure out if `next` can be used like `continue` in other langues! e.g.
> 
> ```brightscript
> for each foo in [1, 2, 3, 4, 5]
>     if foo < 3 then next
>     print foo
> next
> ```
> 
> Is this valid? If so, does it only print `3`, `4`, and `5`?

No!  That's not valid actually.  `next` is a *strict* synonym for `end for` and can only be used where `end for` is used.

**Comment by TwitchBronBron** [comment]
@sjbarag I installed v0.13.0-rc, and it does not appear to have fixed this issue. Here's a test file: 

```javascript
let brs = require('brs');
let lexResult = brs.lexer.Lexer.scan(`
sub countit()
    for each num in [1,2,3]
        print num
    next
end sub`);
let parseResult = brs.parser.Parser.parse(lexResult.tokens);
let errors =[...lexResult.errors, parseResult.errors]; 
//errors should be empty, but instead it contains several errors, 
//one of which states "Expected 'end for' or 'next' to terminate for-loop block"
console.log(errors);
```

**Comment by sjbarag** [comment]
Heya!  You're right - looks like I missed a small change and didn't have an end-to-end test to make sure that worked.  I've added one and should have another RC out for you soon.

**Comment by sjbarag** [comment]
@TwitchBronBron just published `brs@0.13.0-rc.2`, tagged as `rc`.  A simple `npm install brs@rc` should get you a fixed version.  Sorry about that!  :sweat_smile:

**Comment by TwitchBronBron** [comment]
That did the trick. Thanks!
